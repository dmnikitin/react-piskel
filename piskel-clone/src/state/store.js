import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { getFromLocalStorage } from '../helpers/localStorageHandler';
import combined from './reducers/combined';

const localStorageData = getFromLocalStorage();
const store = createStore(combined, localStorageData, composeWithDevTools());

export default store;
