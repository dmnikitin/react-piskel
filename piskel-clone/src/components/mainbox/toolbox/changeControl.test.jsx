import React from 'react'
import { shallow, mount } from 'enzyme';
import ChangeControl from './changeControl';
import store from '../../../state/store';

const props = {
  tool: 'bucket',
  onSetButton: () => { },
}
const wrapperShallow = shallow(<ChangeControl {...props} store={store} />);

describe('changeControls component', () => {
  it('renders correctly', () => {
    expect(wrapperShallow).toMatchSnapshot();
  });
});
