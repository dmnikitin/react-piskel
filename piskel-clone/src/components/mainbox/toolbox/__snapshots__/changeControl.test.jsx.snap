// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changeControls component renders correctly 1`] = `
<ChangeControl
  onSetButton={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  tool="bucket"
/>
`;
